File: java/ex1.java
package jhpark;

import org.neo4j.driver.AuthTokens;
import org.neo4j.driver.Driver;
import org.neo4j.driver.GraphDatabase;
import org.neo4j.driver.Query;

import static org.neo4j.driver.Values.parameters;

public class ex1 implements AutoCloseable {
    private final Driver driver;

    public ex1(String uri, String user, String password) {
        driver = GraphDatabase.driver(uri, AuthTokens.basic(user, password));
    }

    @Override
    public void close() throws RuntimeException {
        driver.close();
    }

    public void printGreeting(final String message) {
        try (var session = driver.session()) {
            var greeting = session.executeWrite(tx -> {
                var query = new Query("CREATE (a:Greeting) SET a.message = $message RETURN a.message + ', from node ' + id(a)", parameters("message", message));
                var result = tx.run(query);
                return result.single().get(0).asString();
            });
            System.out.println(greeting);
        }
    }



    public static void main(String... args) {
    	String NEO4J_URI="neo4j+s://fa216ca0.databases.neo4j.io";
	String NEO4J_USER="neo4j";
	String NEO4J_PASSWORD="1ASLEej5ggsCPSJnqXDHl21FrQCT3b05s0JJQPBx5vY";
        try (var greeter = new ex1(NEO4J_URI, NEO4J_USER, NEO4J_PASSWORD)) {

            greeter.printGreeting("hello, world");
        }
    }
}

File: go/ex1.go
package main

import (
	"fmt"
	"github.com/neo4j/neo4j-go-driver/neo4j"
	"log"
)

func main() {
	// Neo4j connection URL
	uri := "bolt://localhost:7687"
	username := "your_username"
	password := "your_password"

	// Create a Neo4j driver instance
	driver, err := neo4j.NewDriver(uri, neo4j.BasicAuth(username, password, ""))
	if err != nil {
		log.Fatal("Failed to create Neo4j driver:", err)
	}
	defer driver.Close()

	// Create a Neo4j session
	session, err := driver.Session(neo4j.AccessModeWrite)
	if err != nil {
		log.Fatal("Failed to open Neo4j session:", err)
	}
	defer session.Close()

	// Run a Cypher query to create a node
	result, err := session.Run(
		"CREATE (n:Person {name: $name, age: $age}) RETURN n",
		map[string]interface{}{
			"name": "Alice",
			"age":  30,
		})
	if err != nil {
		log.Fatal("Failed to run Cypher query:", err)
	}

	// Process the query result
	for result.Next() {
		record := result.Record()
		node := record.GetByIndex(0) // Get the first item in the record
		fmt.Printf("Created node: %v\n", node)
	}

	if err = result.Err(); err != nil {
		log.Fatal("Query result error:", err)
	}
}
